validator/src/main.rs

```diff
@@ -262,9 +280,16 @@ fn get_rpc_addr(
                         .unwrap()
                         .get_snapshot_hash_for_node(&rpc_peer.id)
                     {
-                        let highest_snapshot_slot_for_node = snapshot_hash
-                            .iter()
-                            .fold(0, |a, (slot, _hash)| a.max(*slot));
+                        let highest_snapshot_slot_for_node =
+                            snapshot_hash.iter().fold(0, |highest_slot, snapshot_hash| {
```

Comment:
Any node that claims to have the highest trusted snapshot hash will do, even if that node itself is untrusted.